### ISSUE TITLE
pyhton 

### ISSUE THREAD

**Issue by gbunkeri** [nltk/nltk#3351]
import string
import nltk
from nltk.corpus import stopwords
from nltk.tokenize import word_tokenize
from nltk.stem import PorterStemmer
from sklearn.feature_extraction.text import TfidfVectorizer

# Ensure you download the required NLTK data files
nltk.download('punkt')
nltk.download('stopwords')

# Sample collection of 16 text documents
documents = [
    "Artificial intelligence is transforming industries.",
    "Machine learning is a subset of artificial intelligence.",
    "Natural language processing enables computers to understand human language.",
    "Deep learning is a type of machine learning.",
    "Data science is an interdisciplinary field.",
    "Big data analytics helps in decision making.",
    "Computer vision allows machines to interpret visual information.",
    "Robotics is a branch of engineering and science.",
    "The future of work involves automation and AI.",
    "Cybersecurity is crucial in the digital age.",
    "Cloud computing provides scalable resources.",
    "Internet of Things connects devices to the internet.",
    "Blockchain technology offers secure transactions.",
    "Augmented reality enhances user experience.",
    "Quantum computing promises unprecedented processing power.",
    "Smart cities utilize technology to improve urban living."
]

# Function to preprocess a single document
def preprocess_document(text):
    # Tokenization
    tokens = word_tokenize(text)

    # Normalization
    normalized_tokens = [token.lower() for token in tokens if token not in string.punctuation]

    # Remove stop words
    stop_words = set(stopwords.words('english'))
    filtered_tokens = [token for token in normalized_tokens if token not in stop_words]

    # Stemming
    stemmer = PorterStemmer()
    stemmed_tokens = [stemmer.stem(token) for token in filtered_tokens]

    # Return preprocessed text
    return ' '.join(stemmed_tokens)

# Preprocess all documents
preprocessed_documents = [preprocess_document(doc) for doc in documents]

# Compute TF-IDF
vectorizer = TfidfVectorizer()
tfidf_matrix = vectorizer.fit_transform(preprocessed_documents)

# Print TF-IDF matrix and feature names
print("TF-IDF Matrix:")
print(tfidf_matrix.toarray())
print("\nFeature Names (Index Terms):")
print(vectorizer.get_feature_names_out())

**Comment by ekaf** [comment]
What is the problem?
